### PR TITLE
Nethack 3.6.0: Added patches and recipe

### PR DESCRIPTION
**Pull request by lich**
So, with tons of experiments I have managed to fix everything and game runs in terminal. The patches are necessary, as even more work is needed to make it work on Arch Linux (sic!).

### CODE REVIEW

File: games-roguelike/nethack/nethack-3.6.0.recipe [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+LICENSE="nethack"
+SOURCE_URI="http://sourceforge.net/projects/nethack/files/nethack/3.6.0/nethack-360-src.tgz"
+CHECKSUM_SHA256="1ade698d8458b8d87a4721444cb73f178c74ed1b6fde537c12000f8edf2cb18a"
+SOURCE_DIR="nethack-$portVersion"
+
+PATCHES="nethack-$portVersion.patchset"
+
+#the x86 is left is somebody wanted to edit it to work under gcc4
+#(it compiles in gcc4, but what is the point if it compiles under gcc2)
+ARCHITECTURES="x86_gcc2 !x86 ?x86_64"
```
Comment (waddlesplash) [inline games-roguelike/nethack/nethack-3.6.0.recipe]: If it compiles in GCC2 *and* GCC4, then get rid of SECONDARY_ARCHITECTURES (+ all `$secondaryArchSuffix`es), and change the `!x86` to `?x86`.

File: games-roguelike/nethack/nethack-3.6.0.recipe [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+DESCRIPTION="NetHack is a single-player roguelike video game originally \
+released in 1987 with ASCII graphics. It is a descendant of an earlier game \
+called Hack (1985), which is a descendant of Rogue (1980)."
+COPYRIGHT="1985-2015 Stichting Mathematisch Centrum and M. Stephenson"
+REVISION="1"
+LICENSE="nethack"
+SOURCE_URI="http://sourceforge.net/projects/nethack/files/nethack/3.6.0/nethack-360-src.tgz"
+CHECKSUM_SHA256="1ade698d8458b8d87a4721444cb73f178c74ed1b6fde537c12000f8edf2cb18a"
+SOURCE_DIR="nethack-$portVersion"
+
```
Comment (waddlesplash) [inline games-roguelike/nethack/nethack-3.6.0.recipe]: Delete this newline.

File: games-roguelike/nethack/nethack-3.6.0.recipe [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+BUILD()
+{
+	./sys/unix/setup.sh ./sys/unix/hints/haiku
+	make PREFIX=$binDir $jobArgs
+}
+
+INSTALL()
+{
+	mkdir $binDir
```
Comment (waddlesplash) [inline games-roguelike/nethack/nethack-3.6.0.recipe]: Add `-p`.

File: games-roguelike/nethack/nethack-3.6.0.recipe [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+		$binDir/nethack/xlogfile \
+		$settingsDir/nethack/
+    #remove first nethack dir to not make a name problems
+	rm -rf $binDir/nethack
+    #replace paths from installation to rights paths in system
+	sed "s/HACKDIR=.*/HACKDIR=\/boot\/system\/settings\/nethack/; \
+		s/HACK=.*/HACK=\/bin\/nethack_bin/" \
+		$binDir/nethacksh/nethack > $binDir/nethack
+	chmod +x $binDir/nethack
+	rm -rf $binDir/nethacksh
```
Comment (waddlesplash) [inline games-roguelike/nethack/nethack-3.6.0.recipe]: Reformat this -- all comments should be indented by one tab, and have a preceding blank line.

**Comment by waddlesplash** [comment]
Also, link to GCI task?

**Comment by lich** [comment]
https://codein.withgoogle.com/dashboard/task-instances/4913154941255680/?sp-page=1

**Comment by waddlesplash** [comment]
Thanks!

**Pull request merged**